modBoot/src/org/aion/Aion.java

```diff
@@ -142,15 +166,42 @@ public static void main(String args[]) {
             zmqThread.start();
         }
 
-        NanoServer rpcServer = null;
-        if (cfg.getApi().getRpc().getActive()) {
-            CfgApiRpc rpcCfg = cfg.getApi().getRpc();
-            rpcServer = new NanoServer(
-                rpcCfg.getIp(),
-                rpcCfg.getPort(),
-                rpcCfg.getCorsEnabled(),
-                rpcCfg.getEnabled(),
-                rpcCfg.getMaxthread());
+        RpcServer rpcServer = null;
+        if(cfg.getApi().getRpc().getActive()) {
+            CfgApiRpc rpcCfg =  cfg.getApi().getRpc();
+
+            Consumer<RpcServerBuilder<? extends RpcServerBuilder<?>>> commonRpcConfig = (rpcBuilder) -> {
+                rpcBuilder.setUrl(rpcCfg.getIp(), rpcCfg.getPort());
+                rpcBuilder.setWorkerPoolSize(rpcCfg.getMaxthread());
+                rpcBuilder.enableEndpoints(rpcCfg.getEnabled());
+
+                if (rpcCfg.getCorsEnabled())
+                    rpcBuilder.enableCorsWithOrigin(rpcCfg.getCorsOrigin());
+
+                CfgSsl cfgSsl = rpcCfg.getSsl();
+                if (cfgSsl.getEnabled())
+                    rpcBuilder.enableSsl(cfgSsl.getCert(), sslPass);
+            };
+            RpcServerVendor rpcVendor = RpcServerVendor.fromString(rpcCfg.getVendor()).orElse(RpcServerVendor.UNDERTOW);
+            switch (rpcVendor) {
+                case NANO: {
+                    NanoRpcServer.Builder rpcBuilder = new NanoRpcServer.Builder();
+                    commonRpcConfig.accept(rpcBuilder);
+                    rpcServer = rpcBuilder.build();
```

Comment:
no try_catch to handle the NullPointerException

Reply:
fixed